README.md

```diff
@@ -5,106 +5,122 @@
 
-> Self identifying base encodings
+> Self-identifying base encodings
 
-Multibase is a protocol for disambiguating the encoding of base-encoded (e.g.,
-base32, base36, base64, base58, etc.) binary appearing in text.
+Multibase is a protocol for disambiguating the "base encoding" used to express
+binary data in text formats (e.g., base32, base36, base64, base58, etc.) from the
+expression alone.
 
 When text is encoded as bytes, we can usually use a one-size-fits-all encoding
 (UTF-8) because we're always encoding to the same set of 256 bytes (+/- the NUL
 byte). When that doesn't work, usually for historical or performance reasons, we
 can usually infer the encoding from the context.
 
-However, when bytes are encoded as text (using a base encoding), the base choice
-of base encoding is often restricted by the context. Worse, these restrictions
-can change based on where the data appears in the text. In some cases, we can
-only use `[a-z0-9]`. In others, we can use a larger set of characters but need a
-compact encoding. This has lead to a large set of "base encodings", one for
-every use-case. Unlike when encoding text to bytes, we can't just standardize
-around a single base encoding because there is no optimal encoding for all
-cases.
+However, when bytes are encoded as text (using a base encoding), the choice of
+base encoding (and alphabet, and other factors) is often restricted by the
+context. Worse, these restrictions can change based on where the data appears in
+the text. In some cases, we can only use `[a-z0-9]`. In others, we can use a
+larger set of characters but need a compact encoding. This has lead to a large
+set of "base encodings", almost one for every use-case. Unlike the case of
+encoding text to bytes, it is impractical to standardize widely around a single
+base encoding because there is no optimal encoding for all cases.
 
-Unfortunately, it's not always clear *what* base encoding is used; that's where
-multibase comes in. It answers the question:
+As data travels beyond its context, it becomes quite hard to ascertain *which*
+base encoding of the many possible ones were used; that's where multibase comes
+in. Where the data has been prefixed before leaving its context behind, it
+answers the question:
 
-> Given data d encoded into text s, what base is it encoded with?
+> Given binary data `d` encoded into text `s`, what base `b` was used to encode it?
+
+To answer this question, a single code point is prepended to `s` at time of
+encoding, which signals in that new context which `b` can be used to reconstruct
+`d`.
 
 ## Table of Contents
 
-- [Format](#format)
-  - [Multibase Table](#multibase-table)
-- [Multibase By Example](#multibase-by-example)
-- [FAQ](#faq)
-- [Implementations:](#implementations)
-- [Disclaimers](#disclaimers)
-- [Maintainers](#maintainers)
-- [Contribute](#contribute)
-- [License](#license)
+- [multibase](#multibase)
+  - [Table of Contents](#table-of-contents)
```

Comment:
Not sure if the table of contents needs the table of contents in it. Though it's totally fine if thats what specs usually do.